share/ansible_navigator/utils/catalog_collections.py

```diff
@@ -112,6 +112,29 @@ def _generate_checksum(file_path: str, relative_path: str) -> Dict:
         }
         return res
 
+    def _path_walk(self, top: Path, top_down: bool = False, follow_links: bool = False):
+        """Emit the file names in a directory tree by walking the tree top-down or bottom-up.
+
+        :param top: The root directory
+        :param top_down: Scan top-down or bottom-up
+        :param follow_links: Visit directories pointed to by symlinks
+        :yields: The top directory, subdirectories and non directories
+        """
+        names = list(top.iterdir())
+
+        dirs = (node for node in names if node.is_dir() is True)
+        nondirs = (node for node in names if node.is_dir() is False)
```

Comment:
Consider only iterating once here. `is_dir()` [calls `stat()`](https://github.com/python/cpython/blob/80e4f262aa27a39abf3fadc19a6323fea4607a8f/Lib/pathlib.py#L1242) and I suspect in a directory with a lot of nodes (and potentially on NFS or something), this can be slow to have to do multiple times. 

Something like this (feel free to pretty it up and expand the if expression):

```python
dirs, nondirs = [], []
for node in names:
    dirs.append(node) if node.is_dir() else nondirs.append(node)
```

Reply:
The path_walk function was removed entirely

Reply:
```
import itertools

numbers = (i for i in range(0, 10))

numbers_1, numbers_2 = itertools.tee(numbers, 2)

assert list(numbers_1) == [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
assert list(numbers_1) == []

assert list(numbers_2) == [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
assert list(numbers_2) == []
```

that's neat :)